packages/client/src/index.ts

```diff
@@ -127,6 +127,24 @@ export function decorate (
   }
 }
 
+const transfersToRemove: string[] = []
+
+/**
+ * Record a transfer to be removed at the next status check
```

Comment:
I think it would be nice to say _why_ it needs to wait until the next status check here. Additionally, we could warn that apps should not rely on `onChange` to fire immediately, which means apps may want to manually hide the transfer until it actually gets removed from storage.